File: typescript/src/modelcontextprotocol/toolkit.ts
import {McpServer} from '@modelcontextprotocol/sdk/server/mcp.js';
import {RequestHandlerExtra} from '@modelcontextprotocol/sdk/shared/protocol.js';
import {Configuration, isToolAllowed} from '../shared/configuration';
import StripeAPI from '../shared/api';
import tools from '../shared/tools';

class StripeAgentToolkit extends McpServer {
  private _stripe: StripeAPI;

  constructor({
    secretKey,
    configuration,
  }: {
    secretKey: string;
    configuration: Configuration;
  }) {
    super({
      name: 'Stripe',
      version: '0.4.0',
      configuration: {
        ...configuration,
        context: {
          ...configuration.context,
          mode: 'modelcontextprotocol',
        },
      },
    });

    this._stripe = new StripeAPI(secretKey, configuration.context);

    const context = configuration.context || {};
    const filteredTools = tools(context);
    // const filteredTools = tools(context).filter((tool) =>
    //   isToolAllowed(tool, configuration)
    // );

    filteredTools.forEach((tool) => {
      this.tool(
        tool.method,
        tool.description,
        tool.parameters.shape,
        async (arg: any, _extra: RequestHandlerExtra<any, any>) => {
          const result = await this._stripe.run(tool.method, arg);
          return {
            content: [
              {
                type: 'text' as const,
                text: String(result),
              },
            ],
          };
        }
      );
    });
  }
}

export default StripeAgentToolkit;

File: python/stripe_agent_toolkit/strands/toolkit.py
"""Stripe Agent Toolkit."""

from typing import List, Optional, Dict
from strands.tools.tools import PythonAgentTool as StrandTool

from ..api import StripeAPI
from ..tools import tools
from ..configuration import Configuration, is_tool_allowed
from .tool import StripeTool
from .hooks import BillingHooks


class StripeAgentToolkit:
    def __init__(
        self, secret_key: str, configuration: Optional[Configuration] = None
    ):
        context = configuration.get("context") if configuration else None

        self._stripe_api = StripeAPI(secret_key=secret_key, context=context)

        filtered_tools = [
            tool for tool in tools if is_tool_allowed(tool, configuration)
        ]

        self._tools = [
            StripeTool(self._stripe_api, tool)
            for tool in filtered_tools
        ]

    def get_tools(self) -> List[StrandTool]:
        """Get the tools in the toolkit."""
        return self._tools

    def billing_hook(
        self,
        type: Optional[str] = None,
        customer: Optional[str] = None,
        meter: Optional[str] = None,
        meters: Optional[Dict[str, str]] = None
    ) -> BillingHooks:
        return BillingHooks(self._stripe_api, type, customer, meter, meters)
